src/snowflake/connector/network.py

```diff
@@ -793,47 +792,34 @@ def fetch(
     ) -> dict[Any, Any]:
         """Carry out API request with session management."""
 
-        class RetryCtx:
+        class RetryCtx(TimeoutBackoffCtx):
             def __init__(
                 self,
-                timeout,
                 _include_retry_params: bool = False,
                 _include_retry_reason: bool = False,
+                **kwargs,
             ) -> None:
-                self.total_timeout = timeout
-                self.timeout = timeout
-                self.cnt = 0
-                self.reason = 0
-                self.sleeping_time = 1
-                self.start_time = get_time_millis()
+                super().__init__(**kwargs)
+                self._reason = 0
                 self._include_retry_params = _include_retry_params
                 self._include_retry_reason = _include_retry_reason
-                # backoff between 1 and 16 seconds
-                self._backoff = DecorrelateJitterBackoff(1, 16)
-
-            def increment_retry(self, reason: int = 0) -> None:
-                """Increment retry count and update the reason for retry
 
-                Args:
-                    reason: HTTP response code that caused retry. Defaults to 0.
-                """
-                self.cnt += 1
-                self.reason = reason
+            @property
+            def retry_reason(self) -> int:
```

Comment:
If these setters and getters don't implement anything special then could we just use  `self.retry_reason` instead of `self._reason`